libazureinit/src/media.rs

```diff
@@ -71,6 +71,20 @@ fn default_preprov_type() -> String {
 pub const PATH_MOUNT_DEVICE: &str = "/dev/sr0";
 pub const PATH_MOUNT_POINT: &str = "/run/azure-init/media/";
 
+const CDROM_VALID_FS: &[&str] = &["iso9660", "udf"];
+
+// Get a mounted device with any filesystem for CDROM
+pub fn get_mount_device() -> Result<String, Error> {
+    if let Some(device) = block_utils::get_mounted_devices()?
+        .iter()
+        .find(|dev| CDROM_VALID_FS.contains(&dev.fs_type.to_str()))
+    {
+        return Ok(device.fs_type.to_string());
```

Comment:
This function should return a list of eligible devices and the caller should iterate through them until they find the device with ovf_env.xml file